web/src/components/storage/ProposalVolumes.jsx

```diff
@@ -0,0 +1,332 @@
+      <FormGroup
+        fieldId="fsType"
+        label="File system type"
+      >
+        <TextInput
+          id="fsType"
+          name="fsType"
+          aria-label="Fs type"
+          value={volume.fsType}
+          label="File system type"
+          isDisabled
+        />
+      </FormGroup>
+      <FormGroup
+        fieldId="minSize"
+        label="Minimum size"
+      >
+        <TextInput
+          id="minSize"
+          name="minSize"
+          aria-label="Min size"
+          value={sizeText(volume.minSize)}
+          label="Minimum Size"
+          isDisabled
+        />
+      </FormGroup>
+      <FormGroup
+        fieldId="maxSize"
+        label="Maximum size"
+      >
+        <TextInput
+          id="maxSize"
+          name="maxSize"
+          aria-label="Max size"
+          value={sizeText(volume.maxSize)}
+          label="Maximum Size"
+          isDisabled
+        />
+      </FormGroup>
+    </Form>
+  );
+};
+
+const GeneralActions = ({ volumes, templates, onVolumesChange }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  const openForm = () => setIsFormOpen(true);
+
+  const closeForm = () => setIsFormOpen(false);
+
+  const acceptForm = (volume) => {
+    closeForm();
+
+    const newVolumes = [...volumes, volume];
+    onVolumesChange(newVolumes);
+  };
+
+  const reset = () => onVolumesChange([]);
+
+  const toggleActions = (status) => setIsOpen(status);
+
+  const closeActions = () => setIsOpen(false);
+
+  const Action = ({ children, ...props }) => (
+    <DropdownItem component="button" {...props}>{children}</DropdownItem>
+  );
+
+  return (
+    <>
+      <Dropdown
+        isOpen={isOpen}
+        onSelect={closeActions}
+        dropdownItems={[
+          <Action
+            key="reset"
+            onClick={reset}
+          >
+            Reset to defaults
+          </Action>,
+          <Action
+            key="add"
+            isDisabled={templates.length === 0}
+            onClick={openForm}
+          >
+            Add file system
+          </Action>
+        ]}
+        toggle={
+          <DropdownToggle toggleVariant="primary" onToggle={toggleActions}>
+            Actions
+          </DropdownToggle>
+        }
+      />
+      <Popup aria-label="Add file system" title="Add file system" isOpen={isFormOpen}>
+        <VolumeForm
+          id="addVolumeForm"
+          templates={templates}
+          onSubmit={acceptForm}
+        />
+        <Popup.Actions>
+          <Popup.Confirm form="addVolumeForm" type="submit">Accept</Popup.Confirm>
+          <Popup.Cancel onClick={closeForm} />
+        </Popup.Actions>
+      </Popup>
+    </>
+  );
+};
+
+const VolumesTable = ({ volumes, onVolumesChange, isLoading }) => {
+  const columns = {
+    mountPoint: "At",
+    details: "Details",
+    size: "Size limits",
+    actions: "Actions"
+  };
+
+  const VolumeRow = ({ id, volume, isLoading }) => {
```

Comment:
If those components are getting all that they need via args, I would move them out of the `VolumesTable`. For me, it is fine for pretty small components or helper functions. But for this kind of stuff, I am not that sure that they should be here.

Reply:
Yes, I am not sure neither. In general, I have only extracted the subcomponents of the main component. But, for a sucomponent itself, I have defined all its subcomponents inside. We need to agree the pattern to follow. So, let's continue with the approach suggested by you: "only use inner components for very small things".